src/vs/workbench/contrib/chat/browser/chatListRenderer.ts

```diff
@@ -291,11 +290,14 @@ export class ChatListItemRenderer extends Disposable implements ITreeRenderer<Ch
 		this.viewModel = viewModel;
 		this._announcedToolProgressKeys.clear();
 
-		if (this._currentThinkingPart) {
-			this._currentThinkingPart.dispose();
-			this._currentThinkingPart = undefined;
+		for (const templateData of this.templateDataByRequestId.values()) {
```

Comment:
This statefulness seems like an issue, why do we keep these around from a previous viewmodel?

Reply:
this was added in for edting previous requests (since sometimes the ids and positions would get moved around on different renders) and is just leveragin those. 

in all the other cases we can get `templateData` from the parameter

Reply:
Ah I remember why we have to do this for editing